src/odemis/odemisd/mdupdater.py

```diff
@@ -284,36 +282,36 @@ def updateLightPower(power, light=light, comp=comp):
 
         return True
 
-    def observeSpectrograph(self, spectrograph, comp):
-        if comp.role != "monochromator":
+    def observeSpectrograph(self, spectrograph, comp_affected):
+        if comp_affected.role != "monochromator":
             return False
 
         if 'slit-monochromator' not in spectrograph.axes:
             logging.info("No 'slit-monochromator' axis was found, will not be able to compute monochromator bandwidth.")
 
-            def updateOutWLRange(pos, sp=spectrograph, comp=comp):
-                wl = sp.position.value["wavelength"]
+            def updateOutWLRange(pos, comp_affected=comp_affected):
+                wl = pos["wavelength"]
                 md = {model.MD_OUT_WL: (wl, wl)}
-                comp.updateMetadata(md)
+                comp_affected.updateMetadata(md)
 
         else:
-            def updateOutWLRange(pos, sp=spectrograph, comp=comp):
+            def updateOutWLRange(pos, sp=spectrograph, comp_affected=comp_affected):
                 width = pos['slit-monochromator']
-                bandwidth = sp.getOpeningToWavelength(width)
+                bandwidth = sp.getOpeningToWavelength(width)  # TODO can you also replace spectrograph here somehow?
                 md = {model.MD_OUT_WL: bandwidth}
-                comp.updateMetadata(md)
+                comp_affected.updateMetadata(md)
 
         spectrograph.position.subscribe(updateOutWLRange, init=True)
         self._onTerminate.append((spectrograph.position.unsubscribe, (updateOutWLRange,)))
 
         return True
 
-    def observeFilter(self, filter, comp):
+    def observeFilter(self, filter, comp_affected):
         # FIXME: If a monochromator + spectrograph, which MD_OUT_WL to pick?
         # update any affected component
-        def updateOutWLRange(pos, fl=filter, comp=comp):
-            wl_out = fl.axes["band"].choices[fl.position.value["band"]]
-            comp.updateMetadata({model.MD_OUT_WL: wl_out})
+        def updateOutWLRange(pos, comp_affected=comp_affected):
+            wl_out = pos["band"]
```

Comment:
Is that then the same in observeSpectrograph? Should I also  change it back there?

Reply:
No, because in `observeSpectrograph` the original code was just `wl = sp.position.value["wavelength"]`, so you can simplify it to `wl = pos["wavelength"]`.
In general, you are just replacing `*.position.value` by `pos`.